docs/docs/reference/connectors/s3.md

```diff
@@ -74,6 +85,39 @@ If you've configured credentials locally already (in your `<RILL_PROJECT_DIRECTO
 
 :::
 
+## IAM Role-Based Authentication
+
+Rill supports AWS IAM role assumption for enhanced security. This method allows Rill to temporarily assume an IAM role to access S3 resources.
+
+### Configuration Parameters
+
+When using IAM role-based authentication, you can configure the following parameters:
+
+- **AWS Role ARN** (`aws_role_arn`) - The Amazon Resource Name (ARN) of the role to assume
+- **AWS Role Session Name** (`aws_role_session_name`) - Optional name for the role session (defaults to "rill-session")
+- **AWS External ID** (`aws_external_id`) - Optional external ID for cross-account access
+
+### Setting Up an IAM Role for Rill
+
+1. **Create an IAM Role**:
+   - In the AWS Management Console, navigate to IAM > Roles > Create role
+   - Select "Another AWS account" as the trusted entity
+   - Enter your AWS account ID or the account ID that Rill will use
```

Comment:
Isn't this supposed to be an AWS service account that Rill manages, not an account managed/created by the user?